### PR TITLE
feat(PTP-1469): metadata class

### PR DESCRIPTION
**Pull request by salimtb**

### CODE REVIEW

**Review by javiergarciavera — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 import Sdk from './src/lib/SDK/sdk.js';
 import Auth from './src/lib/Auth/Auth.js';
-import Metadata from './src/lib/Metadata/Metadata.js';
 import { TEMPLATES } from './src/lib/NFT/constants.js';
+import Metadata from 'src/lib/Metadata/Metadata.js';
```
Comment (MqllR) [inline index.js:10]: Why this change? Relative path is used in the other places...

**Review by danielpavelicconsensys — APPROVED** [review]

File: src/lib/Metadata/Metadata.js [hunk 2]
```diff
@@ -1,134 +1,113 @@
...
+  tokenMetadataSchema,
+} from './metadata.schema.js';
 
 export default class Metadata {
-  IPFS_BASE_URL = 'https://ipfs.io/ipfs/';
+  /**
+   * Create token metadata
+   * @param {string} name Name of the token
+   * @param {string} description description of the token
+   * @param {string} image tokenURI
```
Comment (MqllR) [inline src/lib/Metadata/Metadata.js:13]: image is more a path to your local FS or a link to a remote image, isn't it?

File: src/lib/Metadata/Metadata.js [hunk 3]
```diff
@@ -1,134 +1,113 @@
...
+   * (link to a JSON file describing the token's metadata)
+   * @param {Array} attributes The attributes of the token
+   * @param {string} animation_url Link to the animation such as music, video
+   * @returns string
+   */
+  static tokenLevelMetadata({ name, description, image, attributes, animation_url = null }) {
```
Comment (MqllR) [inline src/lib/Metadata/Metadata.js:19]: Wasn't it supposed to follow OpenSea standard ? https://docs.opensea.io/docs/metadata-standards#metadata-structure

**Review by MqllR — CHANGES_REQUESTED** [review]
`package-lock.json` is broken again ;)

Your implementation just generate a Metadata object, but we should deal with local images as we were doing.

File: src/lib/Metadata/Metadata.js (hunk 2 above)
Comment (MqllR) [reply src/lib/Metadata/Metadata.js:13]: tokenURI or contractURI are a path to the metadata file btw

Comment (salimtb) [reply src/lib/Metadata/Metadata.js:13]: nope it's our tokenURI

Comment (MqllR) [reply src/lib/Metadata/Metadata.js:13]: Maybe the spec is wrong.. `image` is the link to the image.

Look at this https://docs.opensea.io/docs/metadata-standards#metadata-structure

File: src/lib/Metadata/Metadata.js [hunk 4]
```diff
@@ -1,134 +1,135 @@
...
-    return this.upload({
-      metadataObject: Metadata.parseInput(metadataInput),
-      isContract: false,
-    });
-  }
+  /**
+   * Create contract metadata
+   * @param {string} name Name of the contract
+   * @param {string} description description of the contract
+   * @param {string} image contractURI
```
Comment (MqllR) [inline src/lib/Metadata/Metadata.js:70]: Not contractURI, but link to the contract level image

See https://docs.opensea.io/docs/contract-level-metadata

**Review by MqllR — APPROVED** [review]
